src/app/resources/resources.component.ts

```diff
@@ -90,12 +104,82 @@ export class ResourcesComponent implements OnInit, AfterViewInit {
     return (rating * 20) + '%';
   }
 
-  applyResFilter(filterResValue: string) {
-    this.resources.filter = filterResValue.trim().toLowerCase();
+  getRatio(num, dem) {
+    return (num / (num + dem)) * 100;
+  }
+
+  setupList(resourcesRes, ratings) {
+    this.resources.data = resourcesRes.rows.map((r: any) => {
+      const resource = r.doc;
+      const ratingIndex = ratings.findIndex(rating => {
+        return resource._id === rating.item;
+      });
+      if (ratingIndex > -1) {
+        const ratingInfo = this.addRatingToResource(resource._id, ratingIndex, ratings, {
+          rateSum: 0,
+          totalRating: 0,
+          maleRating: 0,
+          femaleRating: 0,
+          hasRated: 0,
+          comment: '',
+          ratingId: '',
+          ratingRev: '',
+          userRating: (ratings[ratingIndex].user.name === this.userService.get().name) ? ratings[ratingIndex].rate : ''
+        });
+        return { ...resource, ...ratingInfo };
+      }
+      return { ...resource,  rateSum: 0, totalRating: 0, maleRating: Math.floor(Math.random() * 101), femaleRating: 0, hasRated: 0  };
```

Comment:
We should replace `Math.random()` with actual data from the ratings database or initialize to 0.